pkg/internal/error/yamlformaterror_test.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package error
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/kubernetes-sigs/kustomize/pkg/constants"
+	yaml "k8s.io/apimachinery/pkg/util/yaml"
+)
+
+var (
+	filepath  = "/path/to/" + constants.KustomizationFileName
+	errorMsg  = "json: line 3: mapping values are not allowed in this context"
+	expectErr = YamlFormatError{
+		Path:     filepath,
+		ErrorMsg: errorMsg,
+	}
+)
+
+func TestYamlFormatError_Error(t *testing.T) {
+	testErr := YamlFormatError{
+		Path:     filepath,
+		ErrorMsg: errorMsg,
+	}
+	if testErr != expectErr {
```

Comment:
Instead of using `expectErr` as a YamlFormatError, can you use an expected error message? 
```
expected := "some expected error message"
if testErr.Error() != expected {
```

Reply:
Done